### PR TITLE
feat: Use unversioned types for new resource creation

### PR DESCRIPTION
**Pull request by ashokponkumar**

### CODE REVIEW

File: internal/apiresourceset/k8sapiresourceset.go [hunk 1]
```diff
@@ -193,18 +194,18 @@ func (k8sAPIResourceSet *K8sAPIResourceSet) ConvertToSupportedVersion(obj runtim
 	fixFn := fixFuncs[kind]
 	versions := clusterSpec.GetSupportedVersions(kind)
 	if versions == nil || len(versions) == 0 {
-		return obj, fmt.Errorf("Kind %s unsupported in target cluster : %+v", kind, obj.GetObjectKind())
+		return nil, fmt.Errorf("Kind %s unsupported in target cluster : %+v", kind, obj.GetObjectKind())
 	}
 	for _, v := range versions {
 		var newobj runtime.Object
+		if kind == "Service" && strings.HasPrefix(v, knativev1.SchemeGroupVersion.Group) {
```
Comment (HarikrishnanBalagopal) [inline internal/apiresourceset/k8sapiresourceset.go:201]: Why revert this? Previously we were parsing and comparing group.

File: internal/apiresource/util.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	corev1 "k8s.io/api/core/v1"
+	core "k8s.io/kubernetes/pkg/apis/core"
+	coreinstall "k8s.io/kubernetes/pkg/apis/core/install"
+)
+
+var (
+	scheme = runtime.NewScheme()
```
Comment (HarikrishnanBalagopal) [inline internal/apiresource/util.go:29]: use a function `GetScheme` instead of global variables.

File: internal/apiresource/util.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+)
+
+func init() {
+	coreinstall.Install(scheme)
```
Comment (HarikrishnanBalagopal) [inline internal/apiresource/util.go:33]: This can be moved inside `GetScheme`.

File: internal/apiresource/util.go [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+func convertCoreObjectVersion(in interface{}, out interface{}) error {
+	err := scheme.Convert(in, out, nil)
```
Comment (HarikrishnanBalagopal) [inline internal/apiresource/util.go:37]: Use `scheme := GetScheme()` instead of global variables.
Also makes it much easier to test.

**Review by HarikrishnanBalagopal — CHANGES_REQUESTED** [review]
Some issues

**Comment by ashokponkumar** [comment]
Will be done along with #404

**Pull request closed without merging**